### PR TITLE
[DOCS-5756] Content connector updates

### PR DESCRIPTION
**Pull request by m-hulbert**

### CODE REVIEW

**Comment by pmartinezga** [comment]
Can you add @nazarethjim as a reviewer? She is the one who mainly developed this connector so maybe she spots something I miss.

File: docs/deployment/upgrade.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ The changelogs for each release are found in the `/config/<service-name>/liquiba
 ## Runtime bundle
 The runtime bundle upgrade scripts are created using a custom approach. 
 
-The [upgrade scripts](https://github.com/Activiti/Activiti/tree/develop/activiti-engine/src/main/resources/org/activiti/db/upgrade) are named slightly differently for the runtime bundle, for example:
+The [upgrade scripts](https://github.com/Activiti/Activiti/blob/7.1.0.M6/activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/) are named slightly differently for the runtime bundle, for example:
```
Comment (pmartinezga) [inline docs/deployment/upgrade.md]: Should the version match the current version `7.1.0.M7` or as it is an example is not needed?

File: docs/modeling/connectors/ootb/README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Out of the box (OOTB) connectors are a set of pre-defined connectors that have b
 * A [Camel connector](../ootb/camel.md) to invoke Camel routes from a process.
 * A [Comprehend connector](../ootb/comprehend.md) using Amazon's Comprehend service to extract insights and relationships from text.
 * A [Docusign connector](../ootb/docusign.md) to send a document to an email address so that it can be signed and stored in Alfresco Content Services.
-* A [DBP connector](../ootb/dbp.md) to create, update and delete content in an instance of Alfresco Content Services (ACS) as part of a process.
+* A [content connector](../ootb/content.md) to create, update and delete content in an instance of Alfresco Content Services (ACS) as part of a process.
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/README.md:12]: Should it start with capital letter (`A [Content connector]`) as the other connectors?

File: docs/modeling/connectors/ootb/README.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ Out of the box (OOTB) connectors are a set of pre-defined connectors that have b
...
 * An [email connector](../ootb/email.md) to automatically send emails as part of a process.
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/README.md:13]: Same here, `Email` starting with capital letter?

File: docs/modeling/connectors/ootb/content.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
+---
+Title: Content connector
+---
+
+# Content connector
+The content connector can be used to execute actions against content in [Alfresco Content Services (ACS)](http://docs.alfresco.com/latest/concepts/welcome.html) as part of a process. The content connector is graphically represented by the Alfresco logo whilst modeling a process.
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: The link to ACS navigates to a page not found, should it be updated?

File: docs/modeling/connectors/ootb/content.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+ <bpmn2:serviceTask id="ServiceTask_00b2568" implementation="content-connector.COPY_CONTENT" />
+``` 
+ 
+### Common input parameters
+The following input parameters are common between content connector actions: 
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- | 
+| `nodeId` | *Required.\** The node ID of the node to operate on in Alfresco Content Services. | String | dioos2a1-2a22-4f42-820d-c2b628751121 |
+| `files` | *Required.\** A [file](../../files.md) uploaded in a process and set as a process variable or uploaded as part of a form or another connector to operate on. | File | |
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: ... uploaded in a `project` and set as... ?

File: docs/modeling/connectors/ootb/content.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+| `search_fields` | *Optional.* Set the list of properties to be returned about the node in the response of the `searchQuery` | Array | id, name |
+
+`*` One of these parameters is required to identify the content to operate on. 
+
+### Common output parameters 
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- | 
+| `nodes` | *Optional.* A list of nodes operated on by the action. | Array | | 
+| `responses` | *Required.* The response from the action. For example the metadata of a node. | JSON | |
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: I think this is an array of HTTP responses, isn't it @nazarethjim? (it is ok keeping JSON type but maybe the description could be more accurate)

File: docs/modeling/connectors/ootb/content.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+| `fields` | *Optional.* Set the list of properties to be returned about the node. | Array | id, name |
+
+### Create a folder
+The `CREATE_FOLDER` action is used to create new folders in the Alfresco Content Services repository.
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- |
+| `folderName` | *Required.* A name for the folder. | String | 2020-shipments | 
+| `autorename` | *Optional.* If set to `true` then a suffix will be added to the folder name if a folder with that name already exists.| Boolean | True | 
+| `nodeType` | *Optional.* The node type for the new content. The default value is `cm:file` | String | cm:file |
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: the deaulf value for a folder is `cm:folder`

File: docs/modeling/connectors/ootb/content.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+| `useFiles` | *Optional.* Set to `true` to include files in the search query.  | Boolean | True |
+| `useFolders`| *Optional.* Set to `true` to include folders in the search query. | Boolean | False | 
+
+### Update content permissions
+The `SET_PERMISSIONS` action is used to update the permissions for files or folders in the Alfresco Content Services repository.
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- |
+| `users` | *Required.* A list of users to update permissions for. | Array | hruser, hradmin | 
+| `role` | *Required.* The roles to assign to the list of `users`. | String | Contributor |
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: I think only one role is possible at each time so I would avoid writing it in plural `The role to assign ...`

File: docs/modeling/connectors/ootb/content.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+| `useFolders`| *Optional.* Set to `true` to include folders in the search query. | Boolean | False | 
+| `include` | *Optional.* Additional information to return about the node. Values include: `properties`, `aspectNames`, `path`, `isLink`, `allowableOperations`, `association`, `isLocked`, `permissions`. | Array | properties, permissions | 
+| `fields` | *Optional.* Set the list of properties to be returned about the node. | Array | id, name |
+
+### Update content tags
+The `UPDATE_TAG` action is used to update tags for files and folders in the Alfresco Content Services repository.
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- |
+| `tag` | *Required.* A list of tags to update for the node. | Array | 2020, orders, online |
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: the parameter name is in plural `tags`

File: docs/modeling/connectors/ootb/content.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+| `id` | The node ID of the node that is created. | String | dioos2a1-2a22-4f42-820d-c2b628751121 | 
+| `nodeType` | The content type that is created. | String | cm:file | 
+| `isFile` | A flag for whether the node is a file. | Boolean | True | 
+| `isFolder` | A flag for whether the node is a folder. | Boolean | False | 
+| `primaryHierarchy` | An array of parent nodes | Array | ioos2a1-2a22-4f42-820d-c2b628751121, 164e0082-5ac1-4afa-a09b-619a8c79f66f | 
+| `properties` | The properties of the created node. | JSON | |
+| `affectedPropertiesBefore` | The properties of the node before the event. | JSON | | 
+| `affectedPropertiesAfter` | The properties of the node after the event. | JSON | | 
+| `aspects` | The aspects of the created node. | Array | | 
+| `aspectNamesBefore` | A list of the aspect names before the event. | Array | versionable |
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: aspects are returned with their prefixed name, for example `cm:versionable` (same below)

File: docs/modeling/connectors/ootb/content.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+
+#### Input parameters
+
+| Parameter | Description | Type | Example |
+| --------  | ----------- | ---- | ------- |
+| `nodeName` | The name of the node that is created. | String | 2020-orders | 
+| `parentFolder` | The node ID of the parent folder. | String | dioos2a1-2a22-4f42-820d-c2b628751121 | 
+| `nodeType` | The content type of the node that is created. | String | cm:file | 
+
+#### Output parameters
```
Comment (pmartinezga) [inline docs/modeling/connectors/ootb/content.md]: Output parameters are common to all the events, so if you prefer you can write it only once as it is done in the connector action with the input parameters. (up to you)

**Review by pmartinezga — CHANGES_REQUESTED** [review]
Just minor issues

File: docs/modeling/connectors/ootb/content.md (hunk 6 above)
Comment (nazarethjim) [reply docs/modeling/connectors/ootb/content.md]: Exactly, it is a list of responses.

File: docs/modeling/connectors/ootb/content.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+| `nodeType` | *Optional.* The node type for the new content. The default value is `cm:file` | String | cm:file | 
+| `include` | *Optional.* Additional information to return about the node. Values include: `properties`, `aspectNames`, `path`, `isLink`, `allowableOperations`, `association`, `isLocked`, `permissions`. | Array | properties, permissions | 
+| `fields` | *Optional.* Set the list of properties to be returned about the node. | Array | id, name |
+
+### Update content
+The `UPDATE_CONTENT` action is used to upload new versions of files to the Alfresco Content Services repository.
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- |
+| `name` | *Optional.* A name for the file. | String | January Orders |
```
Comment (nazarethjim) [inline docs/modeling/connectors/ootb/content.md]: This should include the file extension.

File: docs/modeling/connectors/ootb/content.md [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
+| `include` | *Optional.* Additional information to return about the node. Values include: `properties`, `aspectNames`, `path`, `isLink`, `allowableOperations`, `association`, `isLocked`, `permissions`. | Array | properties, permissions | 
+| `fields` | *Optional.* Set the list of properties to be returned about the node. | Array | id, name |
+
+### Move content
+The `MOVE_CONTENT` action is used to move files or folders to a new location in the Alfresco Content Services repository. If a folder is moved then its children will also move to the new location. 
+
+| Parameter | Description | Type | Example |
+| --------- | ----------- | ---- | ------- |
+| `nodeIdTarget` | *Required.* The node ID of the location to move the content to. | String | 164e0082-5ac1-4afa-a09b-619a8c79f66g |
+| `useFiles` | *Optional.* Set to `true` to include files in the search query.  | Boolean | True |
```
Comment (nazarethjim) [inline docs/modeling/connectors/ootb/content.md:105]:
```suggestion
| `useFiles` | *Optional.* Set to `true` to include only files in the search query.  | Boolean | True |
```

File: docs/modeling/connectors/ootb/content.md [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+| `useFolders`| *Optional.* Set to `true` to include folders in the search query. | Boolean | False|
```
Comment (nazarethjim) [inline docs/modeling/connectors/ootb/content.md:106]:
```suggestion
| `useFolders`| *Optional.* Set to `true` to include only folders in the search query. | Boolean | False| 
```

File: docs/modeling/connectors/ootb/content.md (hunk 8 above)
Comment (nazarethjim) [reply docs/modeling/connectors/ootb/content.md]: Exactly, only one role is possible.

File: docs/modeling/connectors/ootb/content.md [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+
+For a file the event is called `FILE_MOVED`. 
+
+For a folder the event is called `FOLDER_MOVED`.
+
+#### Input parameters
+
+| Parameter | Description | Type | Example |
+| --------  | ----------- | ---- | ------- |
+| `sourceFolder` | The node ID of the folder the node originates from. | String | dioos2a1-2a22-4f42-820d-c2b628751121 |
```
Comment (nazarethjim) [inline docs/modeling/connectors/ootb/content.md]: At the moment, it is not possible to filter with source folder.

**Review by nazarethjim — CHANGES_REQUESTED** [review]
Just minor changes!

File: docs/deployment/upgrade.md (hunk 1 above)
Comment (m-hulbert) [reply docs/deployment/upgrade.md]: It was just in as an example to fix the link but good suggestion, I've updated it!

File: docs/modeling/connectors/ootb/README.md (hunk 2 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/README.md:12]: The others are product names and that's why I've capitalised them, whereas content and email are more generic :-)

File: docs/modeling/connectors/ootb/README.md (hunk 3 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/README.md:13]: As above for the content connector

File: docs/modeling/connectors/ootb/content.md (hunk 5 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Good point!

File: docs/modeling/connectors/ootb/content.md (hunk 4 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Hmm, I've changed that to a static version value but /latest/ should work so I'll fix that separately.

File: docs/modeling/connectors/ootb/content.md (hunk 6 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Ah ok, updated.

File: docs/modeling/connectors/ootb/content.md (hunk 7 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Good spot!

File: docs/modeling/connectors/ootb/content.md (hunk 8 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Updated. I've also added a note that only a single role can be updated at a time.

File: docs/modeling/connectors/ootb/content.md (hunk 9 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Thank you!

File: docs/modeling/connectors/ootb/content.md (hunk 10 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Updated

File: docs/modeling/connectors/ootb/content.md (hunk 11 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: This makes it a lot more readable - good suggestion! 👍

File: docs/modeling/connectors/ootb/content.md (hunk 12 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Is this just for `CREATE_NODE` or for `UPDATE_CONTENT` as well?

File: docs/modeling/connectors/ootb/content.md (hunk 13 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md:105]: Updated all instances as it's in multiple tables - thank you!

File: docs/modeling/connectors/ootb/content.md (hunk 15 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: I'm not sure I understand this one - do you not need the source to fire the event?

Comment (nazarethjim) [reply docs/modeling/connectors/ootb/content.md]: At the moment, this information is not in the ACS event. So we cannot manage it. In the future, we will have it.

File: docs/modeling/connectors/ootb/content.md (hunk 12 above)
Comment (nazarethjim) [reply docs/modeling/connectors/ootb/content.md]: It is for UPDATE_CONTENT

Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: 👍

File: docs/modeling/connectors/ootb/content.md (hunk 15 above)
Comment (m-hulbert) [reply docs/modeling/connectors/ootb/content.md]: Ah ok. I've removed that parameter then until we support it, thanks!

**Review by pmartinezga — APPROVED** [review]

**Pull request merged**